### PR TITLE
Fix: Get index 404

### PR DESCRIPTION
**Pull request by Meldiron**
## What does this PR do?

Get index endpoint in many places expected PHP array, but the actual type was Document. This PR fixes these bugs and makes get index endpoint to work properly.

## Test Plan

- [x] Implement new tests for edge case from issue

## Related PRs and Issues

- https://github.com/appwrite/appwrite/issues/3659

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

Yes 😎

### CODE REVIEW

File: app/controllers/api/databases.php [hunk 1]
```diff
@@ -1675,15 +1675,15 @@ function createAttribute(string $databaseId, string $collectionId, Document $att
         $indexes = $collection->getAttribute('indexes');
 
         // Search for index
-        $indexIndex = array_search($key, array_column($indexes, 'key'));
+        $indexIndex = \array_search($key, \array_map(fn($index) => $index->getAttribute('key'), $indexes));
+
+        \var_dump($indexIndex);
```
Comment (Meldiron) [inline app/controllers/api/databases.php:1680]:
```suggestion
```

**Review by eldadfux — COMMENTED** [review]
We created this method for exactly cases like this, can we use it here?
https://github.com/utopia-php/database/blob/main/src/Database/Document.php#L245

File: app/controllers/api/databases.php [hunk 2]
```diff
@@ -1675,15 +1675,13 @@ function createAttribute(string $databaseId, string $collectionId, Document $att
         $indexes = $collection->getAttribute('indexes');
 
         // Search for index
-        $indexIndex = array_search($key, array_column($indexes, 'key'));
+        $indexIndex = \array_search($key, \array_map(fn($index) => $index->getAttribute('key'), $indexes));
```
Comment (Meldiron) [inline app/controllers/api/databases.php:1678]: (creating discussion so I can resolve it later)

Eldad:
> We created this method for exactly cases like this, can we use it here? https://github.com/utopia-php/database/blob/main/src/Database/Document.php#L245

**Review by christyjacob4 — APPROVED** [review]

**Review by eldadfux — APPROVED** [review]

**Comment by stnguyen90** [comment]
@christyjacob4, @Meldiron what's left on this? I see the Tests failed, but I can't seem to re-run the job. Can you?
